packages/core/utils/src/dml/helpers/entity-builder/define-property.ts

```diff
@@ -214,6 +214,27 @@ export function defineProperty(
     return
   }
 
+  /**
+   * Handling JSON property separately to stringify its default value
+   */
+  if (field.dataType.name === "json") {
+    Property({
+      columnType: "jsonb",
+      type: "any",
+      nullable: field.nullable,
+      fieldName: field.fieldName,
+      /**
+       * MikroORM does not ignore undefined values for default when generating
+       * the database schema SQL. Conditionally add it here to prevent undefined
+       * from being set as default value in SQL.
+       */
+      ...(isDefined(field.defaultValue) && {
+        default: JSON.stringify(field.defaultValue),
+      }),
+    })(MikroORMEntity.prototype, field.fieldName)
```

Comment:
```suggestion
const propertyConfig: PropertyOptions<any> = {  
  columnType: "jsonb",
  type: "any",
  nullable: field.nullable,
  fieldName: field.fieldName,
}

if (isDefined(field.defaultValue)) {
  propertyConfig.default = JSON.stringify(field.defaultValue)
}

Property(propertyConfig)(MikroORMEntity.prototype, field.fieldName)
```

I feel like it would be easier to read that way, wdyt?

Reply:
We have same inline ternary else-where for other properties as well. So it decided to keep it uniformed everywhere.